armi/reactor/converters/uniformMesh.py

```diff
@@ -85,6 +86,317 @@ def converterFactory(globalFluxOptions):
         )
 
 
+class UniformMeshGenerator:
+    """
+    The :py:class:`UniformMeshGenerator <armi.reactor.converters.UniformMeshGenerator>`
+    generates a common axial mesh to for the uniform mesh converter to use. The generation
+    algorithm starts with the simple ``average1DWithinTolerance`` utility function to compute
+    a representative "average" of the assembly meshes in the reactor. It then modifies that
+    mesh to more faithfully represent important material boundaries of fuel and control absorber
+    material.
+
+    The decusping feature is controlled with the case setting ``uniformMeshMinimumSize``. If no
+    value is provided for this setting, the uniform mesh generator will skip the decusping step
+    and just provide the result of ``_computeAverageAxialMesh``.
+    """
+
+    FLOAT_ROUNDING_DECIMALS = 12
+
+    def __init__(self, r, minimumMeshSize=None):
+        """
+        Initialize an object to generate an appropriate common axial mesh to use for uniform mesh conversion.
+
+        Parameters
+        ----------
+        r : :py:class:`Reactor <armi.reactor.reactors.Reactor>` object.
+            Reactor for which a common mesh is generated
+        minimumMeshSize : float, optional
+            Minimum allowed separation between axial mesh points in cm
+            If no miminmum mesh size is provided, no "decusping" is performed
+        """
+        self._sourceReactor = r
+        self.minimumMeshSize = minimumMeshSize
+        self._commonMesh = None
+
+    def generateCommonMesh(self):
+        """
+        Generate a common axial mesh to use.
+
+        Notes
+        -----
+        Attempts to reduce the effect of fuel and control rod absorber smearing
+        ("cusping" effect) by keeping important material boundaries in the
+        common mesh.
+        """
+        self._computeAverageAxialMesh()
+        if self.minimumMeshSize is not None:
+            self._decuspAxialMesh()
+
+    def _computeAverageAxialMesh(self):
+        """
+        Computes an average axial mesh based on the core's reference assembly.
+
+        Notes
+        -----
+        This iterates over all the assemblies in the core and collects all assembly meshes
+        that have the same number of fine-mesh points as the `refAssem` for the core. Based on
+        this, the proposed uniform mesh will be some average of many assemblies in the core.
+        The reason for this is to account for the fact that multiple assemblies (i.e., fuel assemblies)
+        may have a different mesh due to differences in thermal and/or burn-up expansion.
+        """
+        src = self._sourceReactor
+        refAssem = src.core.refAssem
+
+        refNumPoints = len(src.core.findAllAxialMeshPoints([refAssem])) - 1
```

Comment:
add a comment above explaning the -1 is for non-zero points or calculate it the same way (with `[:1]`)